native/cocos/2d/renderer/UIMeshBuffer.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include <cocos/base/TypeDef.h>
+#include <vector>
+#include <renderer/gfx-base/GFXDef-common.h>
+
+namespace cc {
+struct MeshBufferLayout {
+    index_t byteOffset;
+    index_t vertexOffset;
+    index_t indexOffset;
+    index_t dirtyMark;
+    index_t floatsPerVertex;
+};
+
+class UIMeshBuffer {
+public:
+    UIMeshBuffer();
+    ~UIMeshBuffer();
+
+    inline float_t* getVData() { return _vData; }
+    void setVData(float_t* vData);
+    inline uint16_t* getIData() { return _iData; }
+    void setIData(uint16_t* iData);
+
+public:
+    void initialize(gfx::Device* device, ccstd::vector<gfx::Attribute*>&& attrs, index_t vFloatCount, index_t iCount);
+    void reset();
+    void destroy();
+    void setDirty();
+    void uploadBuffers();
+
+    void syncSharedBufferToNative(index_t* buffer);
```

Comment:
- I don't think use `index_t *` is needed. Just use `index_t`.
- Use `uint32_t` instead?

Reply:
The reason of using index_t* is we need a shared array to store some values.
It does need to modify to uint32_t.